### PR TITLE
ISPN-11709 The MassIndexer reports wrong status for local caches

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-11709

### CODE REVIEW

File: query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java [hunk 1]
```diff
@@ -134,7 +129,7 @@ public boolean isRunning() {
             IndexWorker indexWork = new IndexWorker(cache.getName(), indexedTypes, skipIndex, null);
             futures.add(executor.submitConsumer(indexWork, TRI_CONSUMER));
             CompletableFuture<Void> compositeFuture = CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
-            return compositeFuture.whenCompleteAsync(flushIfNeeded, localExecutor);
+            return compositeFuture.whenComplete(flushIfNeeded);
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java:132]: Does `flushIfNeeded` not block?

Comment (gustavonalle) [reply query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java:132]: yes

Comment (gustavonalle) [reply query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java:132]: This change was not necessary either, I will rollback

Comment (wburns) [reply query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java:132]: I always enjoy removing usages of blocking executor if we can :)

**Comment by pruivo** [comment]
integrated! thanks @gustavonalle !

**Pull request merged**